youtube_dl/extractor/hotstar.py

```diff
@@ -99,3 +100,59 @@ def _real_extract(self, url):
             'episode_number': int_or_none(video_data.get('episodeNumber')),
             'series': video_data.get('contentTitle'),
         }
+
+class HotStarBaseIE(InfoExtractor):
+    @classmethod
+    def _extract_url_info(cls, url):
+        mobj = re.match(cls._VALID_URL, url)
+        return mobj.group('series_id'), mobj.group('playlist_id'), mobj.group('playlist_title')
+
+    def _extract_from_json_url(self, series_id, playlist_title, video ):
+
+        picture_url = video.get('urlPictures');
+        thumbnail = 'http://media0-starag.startv.in/r1/thumbs/PCTV/%s/%s/PCTV-%s-hs.jpg' % ( picture_url[-2:], picture_url, picture_url )
+
+        episode_title = video.get('episodeTitle')
+        episode_title = episode_title.lower().replace(' ', '-')
+        url = "http://www.hotstar.com/tv/%s/%s/%s/%s" % (playlist_title, series_id, episode_title, video.get('contentId'))
+
+        info_dict = {
+            'id': video.get('contentId'),
+            'title': video.get('episodeTitle'),
+            'description': video.get('longDescription'),
+            'thumbnail' : thumbnail,
+            'url' : url,
+            '_type' : 'url',
+        }
+        return info_dict
+
+class HotStarPlaylistIE(HotStarBaseIE):
+    IE_NAME = 'hotstar:playlist'
+    _VALID_URL = r'https?://(?:www\.)?hotstar\.com/tv/(?P<playlist_title>.+)/(?P<series_id>\d+)/episodes/(?P<playlist_id>\d{1,})'
+
+    _TESTS = [{
+        'url': 'http://www.hotstar.com/tv/pow-bandi-yuddh-ke/10999/episodes/10856/9993',
+        'info_dict': {
+            'id': '10856',
+            'title': 'pow-bandi-yuddh-ke',
+        },
+        'playlist_mincount': 0,
+    }, {
+        'url': 'http://www.hotstar.com/tv/pow-bandi-yuddh-ke/10999/episodes/10856/9993',
+        'only_matching': True,
+    }]
+
+    def _real_extract(self, url):
+        series_id, playlist_id, playlist_title = self._extract_url_info(url)
+
+        collection = self._download_json(
+            "http://search.hotstar.com/AVS/besc?action=SearchContents&appVersion=5.0.39&channel=PCTV&moreFilters=series:%s;&query=*&searchOrder=last_broadcast_date+desc,year+asc,title+asc&type=EPISODE" % playlist_id,
```

Comment:
Tried to use it but this seems to be encoded. server is not able to accept encoded params. let me know if there is way not to encode this url